internal/coreinternal/processor/filterspan/durationmatcher.go

```diff
@@ -0,0 +1,50 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterspan // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterspan"
+
+import (
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterconfig"
+)
+
+type SpanDurationMatcher struct {
+	Operation string
+	Value     int64
+}
+
+func (ma SpanDurationMatcher) Match(span ptrace.Span) bool {
+	duration := span.EndTimestamp().AsTime().Unix() - span.StartTimestamp().AsTime().Unix()
+
+	switch ma.Operation {
+	case ">":
```

Comment:
The mismatch between the validation code and the reality here means that we have a gap in unit tests. Please, add tests exercising all operators.

Reply:
This relates to a previous comment that other operators exist. The fact that this mismatch (docs vs. code) wasn't uncovered until the review cycle means we have a possible gap in the unit tests: if the other operators should be supported, we should have tests for them.

Reply:
OH, I see.  Will do!